src/CMakeLists.txt

```diff
@@ -291,7 +305,57 @@ if (BUILDSWIGNODE)
     ${NODE_ROOT_DIR}/include/deps/uv/include
   )
   macro(createpackagejson)
+    if (NPM)
+      # Dependency to generate wrappers before packing, skip modules that depend on 3rd party libs
+      if (NOT DEFINED reqlibname)
+        set (reqlibname "")
+      endif ()
+      if (reqlibname STREQUAL "" OR reqlibname MATCHES "mraa")
+        # Grab all module sources and headers, also add sources to gyp list
+        file (GLOB srcfiles RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}"
```

Comment:
It is possible to get the SOURCES property for both the base library and NODE wrapper library targets which would provide a finite list of src/hdr files to use.